src/Symfony/Component/Messenger/Worker.php

```diff
@@ -218,6 +224,30 @@ private function ack(): bool
         return (bool) $acks;
     }
 
+    private function rateLimit(string $transportName): void
+    {
+        if (null === $this->rateLimiterLocator) {
+            return;
+        }
+
+        if (!$this->rateLimiterLocator->has($transportName)) {
+            return;
+        }
+
+        /** @var LimiterInterface $rateLimiter */
+        $rateLimiter = $this->rateLimiterLocator->get($transportName)->create();
+        if ($rateLimiter->consume()->isAccepted()) {
+            return;
+        }
+
+        $this->logger?->info('Transport {transport} is being rate limited, waiting for token to become available...', [
+            'transport' => $transportName,
+        ]);
+
+        $this->dispatchEvent(new WorkerRateLimitedEvent($rateLimiter, $transportName));
+        $rateLimiter->reserve()->wait();
```

Comment:
Does it mean that it blocks the worker from consuming other messages?

Reply:
That is correct. The rate limiter is configured for a transport, so the docs should mention that if you do not want to block other transports, you should use a dedicated worker.

Reply:
Indeed, this must be front and center as this is a significant limitation IMO.

Reply:
Added the caution to the new docs, see https://github.com/symfony/symfony-docs/pull/17203